### PR TITLE
Remove sm_stringutil.

### PR DESCRIPTION
**Pull request by dvander**
Well, these patches don't remove it completely, but they reduce it to just two functions and pretty minimal headers. All other calls get replaced with AMTL.

### CODE REVIEW

File: core/MenuManager.cpp [hunk 1]
```diff
@@ -446,7 +446,7 @@ IMenuPanel *MenuManager::RenderMenu(int client, menu_states_t &md, ItemOrder ord
 		char text[50];
 		if (!logicore.CoreTranslate(text, sizeof(text), "%T", 2, NULL, "No Vote", &client))
 		{
-			UTIL_Format(text, sizeof(text), "No Vote");
+			ke::SafeSprintf(text, sizeof(text), "No Vote");
```
Comment (KyleSanderson) [inline core/MenuManager.cpp]: nit: all changes in this file can be ke::SafeStrcpy.

File: core/NextMap.h [hunk 2]
```diff
@@ -33,16 +33,17 @@
...
+#include <amtl/am-string.h>
 
 struct MapChangeData
 {
 	MapChangeData(const char *mapName, const char *changeReason, time_t time)
 	{
-		UTIL_Format(m_mapName, sizeof(m_mapName), mapName);
-		UTIL_Format(m_changeReason, sizeof(m_changeReason), changeReason);
+		ke::SafeSprintf(m_mapName, sizeof(m_mapName), mapName);
+		ke::SafeSprintf(m_changeReason, sizeof(m_changeReason), changeReason);
```
Comment (KyleSanderson) [inline core/NextMap.h]: we could ke::SafeStrcpy here to avoid reformatting the string, or "%s" it. Alternatively, ke::AString.

**Comment by KyleSanderson** [comment]
I don't see any regressions, just a bunch of pre-existing nits. :shipit:

**Comment by dvander** [comment]
Merged via command-line with nits picked.

**Pull request closed without merging**